src/types.rs

```diff
@@ -118,6 +119,11 @@ impl KeyAuthorization {
     pub fn dns_value(&self) -> String {
         base64::encode_config(digest(&SHA256, self.0.as_bytes()), URL_SAFE_NO_PAD)
     }
+    
+    /// Get the SHA256 digest of the key authorization
+    pub fn tls_value(&self) -> Vec<u8> {
```

Comment:
IIRC the value here is the same for ALPN and HTTP, right? So maybe it should just be called `value`? Also I think we could have this return the `digest()` directly as an `impl AsRef<[u8]>`, avoiding the extra allocation.

Reply:
https://www.rfc-editor.org/rfc/rfc8737.txt (end of sec.3), so it should be the result of the SHA-256.
I am not sure how it is possible to return the `digest()` value directly as `AsRef<[u8]>` without unsafe code because it will be considered a local variable, and it is impossible to return its reference. I would appreciate it if you give me a clue to do it.

Reply:
`digest()` returns a `Digest` which implements `AsRef<[u8]>`, so you can just return the `Digest` directly (but I don't want it to appear in the type signature directly to prevent coupling with the *ring* API).

Reply:
> `digest()` returns a `Digest` which implements `AsRef<[u8]>`, so you can just return the `Digest` directly (but I don't want it to appear in the type signature directly to prevent coupling with the _ring_ API).

Gotcha, I thought you were talking about wrapping the `as_ref`. What do you think about implementing it `as_bytes`? Then we internally call it for the DNS one.

Reply:
I think the `as_` prefix is usually used for accessors that don't do much actual work, so I don't that name is a good fit here  - could be `to_bytes()` though, and using it inside `dns_value()` makes sense to me!